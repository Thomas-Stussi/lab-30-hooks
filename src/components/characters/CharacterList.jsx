/* eslint-disable max-len */
import React from 'react';
import PropTypes from 'prop-types';
import CharacterItem from './CharacterItem';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map(character => (
    <li key={character.name}>
      <a href={`/${character.name}`}>
        <CharacterItem {...character} />
      </a>
    </li>
  ));

  return (
    <ul data-testid="characters">
      {characterElements}
    </ul>
  );
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired
  })).isRequired
};

export default CharacterList;
