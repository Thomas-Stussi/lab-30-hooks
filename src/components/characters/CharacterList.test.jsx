/* eslint-disable max-len */
import React from 'react';
import { render } from '@testing-library/react';
import CharacterList from './CharacterList';

describe('List Component', () => {
  const characterArray = [{ name: 'test1', image: 'https://www.placecage.com/c/140/200' }, { name: 'test2', image: 'https://www.placecage.com/c/140/200' }];

  it('renders List component', () => {
    const { asFragment } = render(<CharacterList characters={characterArray} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
