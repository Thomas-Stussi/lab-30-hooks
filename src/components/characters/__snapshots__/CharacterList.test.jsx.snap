// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Component renders List component 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  >
    <li>
      <a
        href="/test1"
      >
        <img
          src="https://www.placecage.com/c/140/200"
        />
        <h3>
          test1
        </h3>
      </a>
    </li>
    <li>
      <a
        href="/test2"
      >
        <img
          src="https://www.placecage.com/c/140/200"
        />
        <h3>
          test2
        </h3>
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
